src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -341,13 +341,21 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
         // assumes all same case.        
         private bool ArgumentMatch(string arg, string formal)
         {
-            if (arg[0] != '/' && arg[0] != '-')
+            if (arg[0] != '-' )
             {
                 return false;
             }
 
+            // Full name format, eg: --assembly
+            if (arg.Length > 1 && arg[1] == '-')
+            {
+                arg = arg.Substring(2);
+                return arg == formal;
```

Comment:
We should use string.Equals for the comparison and use StringComparison enum to control how comparison happens (eg. ignore case).